tests/foreman/api/test_repository.py

```diff
@@ -639,8 +641,45 @@ def test_negative_create_checksum(self):
 
         :CaseImportance: Critical
         """
-        with self.assertRaises(HTTPError):
-            entities.Repository(checksum_type=gen_string('alpha')).create()
+        for downloadpolicy in 'background', 'immediate':
+            with self.assertRaises(HTTPError):
+                entities.Repository(checksum_type=gen_string('alpha'),
+                                    download_policy=downloadpolicy).create()
+
+    @tier1
+    @run_only_on('sat')
+    def test_negative_create_checksum_with_on_demand_policy(self):
+        """Attempt to create repository with checksum and on_demand policy.
+
+        :id: de8b157c-ed62-454b-94eb-22659ce1158e
+
+        :expectedresults: A repository is not created and error is raised.
+
+        :CaseImportance: Critical
+        """
+        for checksum_type in 'sha1', 'sha256':
+            with self.assertRaises(HTTPError):
+                entities.Repository(checksum_type=checksum_type).create()
+
+    @tier1
+    @run_only_on('sat')
+    def test_negative_update_checksum_with_on_demand_policy(self):
+        """Attempt to update the on_demand downloadpolicy on a already created repository with
+        checksum.
+
+        :id: 5bfaef4f-de66-42a0-8419-b86d00ffde6f
+
+        :expectedresults: A repository is not updated and error is raised.
+
+        :CaseImportance: Critical
+        """
+        for downloadpolicy in 'background', 'immediate':
```

Comment:
I feel that testing both loops in more than one test case is redundant. You can test them in positive scenario and take one example for negative

Reply:
@oshtaier  It's good to cover more here because a single change in download_policy could impact at multiple others.

Reply:
also we could separate as per download policy

Reply:
right, but if we have test case that already validate all these combinations and passed or failed, we do not need to do the same things in other test cases, as we already know the result